### ISSUE TITLE
Clang CPU type can be any string

### ISSUE THREAD

**Issue by mlund** [llvm-mos/llvm-mos#213]
Clang's `-mcpu` option currently accepts any cpu type, e.g. the following compiles with no warning:
``` bash
mos-c64-clang test.c -o test.prg -mcpu=foo
```
I had expected an error as `foo` is not a known cpu type, i.e. only `mos6502`, `mos65c02` etc. should be accepted.

_Perhaps_ related:
1. adding `-mcpu=mos..` to `clang.cfg` for specific targets is silently ignored.
2. `clang --config clang.cfg` silently accepts any (non-existent) file
